### PR TITLE
refactor: move `hover` into a proper `@media` hover

### PR DESCRIPTION
**Pull request by sonukapoor**
This commit moves the contributor hover into the `@media(hover)`
query. This will help to identify if the user's primary input mechanism
can hover over elements.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: aio/src/styles/2-modules/_contributor.scss [hunk 1]
```diff
@@ -52,16 +52,18 @@ aio-contributor {
   transition: all .3s;
   perspective: 800px;
 
-  &:hover {
-    transform: translate3d(0,-3px,0);
-    box-shadow: 0 8px 8px rgba(10, 16, 20, 0.24), 0 0 8px rgba(10, 16, 20, 0.12);
+  @media(hover:hover) {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_contributor.scss:55]: For consistency:

```suggestion
  @media (hover) {
```

**Review by gkalpak — CHANGES_REQUESTED** [review]
Thx, @sonukapoor! Can you please change the commit message to `fix(docs-infra): ...`?

File: aio/src/styles/2-modules/_contributor.scss (hunk 1 above)
Comment (sonukapoor) [reply aio/src/styles/2-modules/_contributor.scss]: @gkalpak Done. Please re-review. Thanks

File: aio/src/styles/2-modules/_contributor.scss [hunk 2]
```diff
@@ -52,16 +52,18 @@ aio-contributor {
   transition: all .3s;
   perspective: 800px;
 
-  &:hover {
-    transform: translate3d(0,-3px,0);
-    box-shadow: 0 8px 8px rgba(10, 16, 20, 0.24), 0 0 8px rgba(10, 16, 20, 0.12);
+  @media(hover) {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_contributor.scss:55]:
```suggestion
  @media (hover) {
```

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/src/styles/2-modules/_contributor.scss (hunk 2 above)
Comment (sonukapoor) [reply aio/src/styles/2-modules/_contributor.scss]: Done.

**Review by gkalpak — APPROVED** [review]
Thx for making the changes, @sonukapoor. Can you please squash the commits into one and make sure CI is happy.

**Comment by sonukapoor** [comment]
@gkalpak Can you review this again? I think this is good to be merged?

**Pull request closed without merging**